tools/generate_tables_sql_and_testNames.php

```diff
@@ -23,95 +26,118 @@
-            //generate specific column definitions for specific types of HTML elements
-            default:
-                if($bits[1] == "") {
-                    continue 2;
-                }
-                if($bits[0]=="select"){
-                    $bits[0]=enumizeOptions($bits[3], isset($tablename) ? $tablename : null, $bits[1]);
-                } else if($bits[0]=="selectmultiple"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="textarea"){
-                    $bits[0]="text";
-                } else if($bits[0]=="text"){
-                    $bits[0]="varchar(255)";
-                } else if($bits[0]=="checkbox") {
-                    $bits[0]="varchar(255)";
-                } else if ($bits[0]=="static") {
-                    $bits[0]="varchar(255)";
-                }
 
-                $bits[2]=htmlspecialchars($bits[2]);
-                $output.="`$bits[1]` $bits[0] default NULL,\n";
+        // generate specific column definitions for specific HTML elements
+        default:
+            if ($bits[1] == "") {
+                continue 2;
+            }
+            if ($bits[0]== "select") {
+                $bits[0] =enumizeOptions(
+                    $bits[3],
+                    $tablename ?? null,
+                    $bits[1]
+                );
+            } elseif ($bits[0]== "selectmultiple") {
+                $bits[0] = "varchar(255)";
+            } elseif ($bits[0]== "textarea") {
+                $bits[0] = "text";
+            } elseif ($bits[0]== "text") {
+                $bits[0] = "varchar(255)";
+            } elseif ($bits[0]== "checkbox") {
+                $bits[0] = "varchar(255)";
+            } elseif ($bits[0] == "static") {
+                $bits[0] = "varchar(255)";
+            }
+
+            $bits[2] =htmlspecialchars($bits[2]);
+            $output .= "`$bits[1]` $bits[0] default NULL,\n";
         }
     }
-
-    $output.="PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
+    $output .= "PRIMARY KEY  (`CommentID`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8;\n";
     print "Filename: $filename\n";
-    $fp=fopen($filename, "w");
+    $fp =fopen($filename, "w");
     fwrite($fp, $output);
-    fwrite($fp, "REPLACE INTO test_names (Test_name, Full_name, Sub_group) VALUES ('$tablename', '$title', 1);\n");
-    foreach($pages as $pageNo => $page) {
-        fwrite($fp, "INSERT INTO instrument_subtests (Test_name, Subtest_name, Description) VALUES ('$tablename', '" . $tablename  . "_page" . ($pageNo+1) . "', '$page');\n");
+    $sql = "REPLACE INTO test_names (Test_name, Full_name, Sub_group) 
+VALUES ('$tablename', '$title', 1);";
+
+    fwrite($fp, $sql);
+    foreach ($pages as $pageNo => $page) {
+        unset($sql);
+        $sql = "INSERT INTO instrument_subtests (
+  Test_name, 
+  Subtest_name, 
+  Description
+) VALUES ('%s', '%s_page%s, '%s');";
+        fwrite($fp, sprintf($sql, $tableName, $tableName, $pageNo + 1, $page));
```

Comment:
This syntax is weird. I don't see why this is being converted to a sprintf, but given that it is why is it using `fwrite( ... sprintf)` instead of `fprintf`?